src/psyclone/transformations.py

```diff
@@ -356,7 +355,110 @@ def apply(self, node, options=None):
             self.omp_nogroup = current_nogroup
 
 
-class OMPDeclareTargetTrans(Transformation):
+class MarkRoutineForGPUMixin:
+
+    def validate_it_can_run_on_gpu(self, node, options):
+        '''
+        Check that an this node can be marked as available to be called on
```

Comment:
"...that the supplied node..."
Please also document the arguments and the returns and raises.

Reply:
Done. I added (or left) the same raises in the subclasses of the mixin, because even though they don't raise it directly, it is probably the public interface that the user will see.